src/core/dfn-panel.runtime.js

```diff
@@ -97,10 +140,67 @@ function displayPanel(dfn, panel, { x, y }) {
     const caret = panel.querySelector(".caret");
     caret.style.left = `${newCaretOffset}px`;
   }
+
+  // If we are displaying because of a keydown,
+  // we want to trap focus
+  if (eventType === "keydown") {
```

Comment:
This doesn't work for screen readers. Screen readers fire a synthetic click event when activating links so none of this logic will work. You can't branch on event type for this logic. Why not move focus for a mouse user too? You can dismiss the dialog when they click anywhere else on the screen.